paddle/fluid/framework/ir/graph_pattern_detector.cc

```diff
@@ -1056,12 +1056,17 @@ PDNode *patterns::FC::operator()(paddle::framework::ir::PDNode *x,
   }
 }
 
-PDNode *patterns::FCMKLDNN::operator()(paddle::framework::ir::PDNode *x,
-                                       bool with_bias) {
-  // Create shared nodes.
-  x->assert_is_op_input("fc", "Input");
-
+PDNode *patterns::FCMKLDNN::operator()(bool with_residual_data) {
   auto *fc_op = pattern->NewNode(fc_repr())->assert_is_op("fc");
+
+  if (!with_residual_data) {
+    fc_op->assert_more([&](Node *x) {
+      if (!HasOutput(x, "ResidualData") ||
+          x->Op()->Output("ResidualData").size() == 0)
+        return true;
+      return false;
+    });
+  }
```

Comment:
Could you keep these 2 ifs close to each other?
It will make it easier to read.
```
if(with_residual_data) {
...
} else {
...
}

Reply:
It might work